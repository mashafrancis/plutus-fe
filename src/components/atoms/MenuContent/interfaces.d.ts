export interface MenuContentProps {
	name: string;
	photo: string;
}
