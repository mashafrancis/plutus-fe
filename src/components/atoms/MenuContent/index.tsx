import { ChangeEvent, useContext } from 'react';
// components
import { ComponentContext } from '@context/ComponentContext';
import { MenuTab, MenuTabs } from '@components/atoms';
import { DashboardMenus } from '@components/molecules/MenuRoutes';

const MenuContent = (): JSX.Element => {
	const { selectedIndex, setSelectedIndex } = useContext(ComponentContext);

	const handleOnChange = (
		event: ChangeEvent<HTMLDivElement>,
		value: number,
	) => {
		setSelectedIndex(value);
	};

	const a11yProps = (index: number | string) => {
		return {
			id: `menu-tab-${index}`,
			'aria-controls': `menu-tabpanel-${index}`,
		};
	};

	return (
		<MenuTabs
			style={{ padding: 20 }}
			value={selectedIndex}
			// @ts-expect-error
			onChange={handleOnChange}
			orientation="vertical"
			scrollButtons={false}
			textColor="primary"
			aria-label="menu tabs"
			visibleScrollbar={false}
		>
			{DashboardMenus.slice(0, 6).map((item) => (
				<MenuTab
					key={item.primaryText}
					label={item.primaryText}
					icon={item.icon}
					{...a11yProps(selectedIndex)}
				/>
			))}
		</MenuTabs>
	);
};

export default MenuContent;
