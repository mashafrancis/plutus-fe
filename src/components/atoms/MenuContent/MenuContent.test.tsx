// react libraries
import { render } from '@testing-library/react';
// component
import MenuContent from './index';

describe('MenuContent component', () => {
	it('should render correctly', () => {
		const { asFragment } = render(<MenuContent />);
		expect(asFragment()).toMatchSnapshot();
	});
});
