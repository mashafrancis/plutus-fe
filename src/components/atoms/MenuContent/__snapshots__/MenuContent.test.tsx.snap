// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuContent component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiTabs-root WithStyles(ForwardRef(Tabs))-root-1 MuiTabs-vertical"
    style="padding: 20px;"
  >
    <div
      class="MuiTabs-scroller MuiTabs-fixed"
      style="overflow: hidden;"
    >
      <div
        aria-label="menu tabs"
        class="MuiTabs-flexContainer MuiTabs-flexContainerVertical"
        role="tablist"
      >
        <button
          aria-controls="menu-tabpanel-0"
          aria-selected="true"
          class="MuiButtonBase-root MuiTab-root Component-root-3 MuiTab-textColorPrimary Mui-selected Component-selected-4 MuiTab-labelIcon"
          id="menu-tab-0"
          role="tab"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M13 14v6c0 .55.45 1 1 1h6c.55 0 1-.45 1-1v-6c0-.55-.45-1-1-1h-6c-.55 0-1 .45-1 1zm-9 7h6c.55 0 1-.45 1-1v-6c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1v6c0 .55.45 1 1 1zM3 4v6c0 .55.45 1 1 1h6c.55 0 1-.45 1-1V4c0-.55-.45-1-1-1H4c-.55 0-1 .45-1 1zm12.95-1.6L11.7 6.64c-.39.39-.39 1.02 0 1.41l4.25 4.25c.39.39 1.02.39 1.41 0l4.25-4.25c.39-.39.39-1.02 0-1.41L17.37 2.4c-.39-.39-1.03-.39-1.42 0z"
              />
            </svg>
            Analytics
          </span>
        </button>
        <button
          aria-controls="menu-tabpanel-0"
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root Component-root-3 MuiTab-textColorPrimary MuiTab-labelIcon"
          id="menu-tab-0"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M17.66 8L12 2.35 6.34 8C4.78 9.56 4 11.64 4 13.64s.78 4.11 2.34 5.67 3.61 2.35 5.66 2.35 4.1-.79 5.66-2.35S20 15.64 20 13.64 19.22 9.56 17.66 8zM6 14c.01-2 .62-3.27 1.76-4.4L12 5.27l4.24 4.38C17.38 10.77 17.99 12 18 14H6z"
              />
            </svg>
            Water Cycles
          </span>
        </button>
        <button
          aria-controls="menu-tabpanel-0"
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root Component-root-3 MuiTab-textColorPrimary MuiTab-labelIcon"
          id="menu-tab-0"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.54 5.54L13.77 7.3 12 5.54 10.23 7.3 8.46 5.54 12 2zm2.92 10l-1.76-1.77L18.46 12l-1.76-1.77 1.76-1.77L22 12zm-10 2.92l1.77-1.76L12 18.46l1.77-1.76 1.77 1.76L12 22zm-2.92-10l1.76 1.77L5.54 12l1.76 1.77-1.76 1.77L2 12z"
              />
              <circle
                cx="12"
                cy="12"
                r="3"
              />
            </svg>
            Environment
          </span>
        </button>
        <button
          aria-controls="menu-tabpanel-0"
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root Component-root-3 MuiTab-textColorPrimary MuiTab-labelIcon"
          id="menu-tab-0"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 1L3 5v6c0 5.55 3.84 10.74 9 12 5.16-1.26 9-6.45 9-12V5l-9-4zm0 10.99h7c-.53 4.12-3.28 7.79-7 8.94V12H5V6.3l7-3.11v8.8z"
              />
            </svg>
            Quality Control
          </span>
        </button>
        <button
          aria-controls="menu-tabpanel-0"
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root Component-root-3 MuiTab-textColorPrimary MuiTab-labelIcon"
          id="menu-tab-0"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15 9H9v6h6V9zm-2 4h-2v-2h2v2zm8-2V9h-2V7c0-1.1-.9-2-2-2h-2V3h-2v2h-2V3H9v2H7c-1.1 0-2 .9-2 2v2H3v2h2v2H3v2h2v2c0 1.1.9 2 2 2h2v2h2v-2h2v2h2v-2h2c1.1 0 2-.9 2-2v-2h2v-2h-2v-2h2zm-4 6H7V7h10v10z"
              />
            </svg>
            Energy Usage
          </span>
        </button>
        <button
          aria-controls="menu-tabpanel-0"
          aria-selected="false"
          class="MuiButtonBase-root MuiTab-root Component-root-3 MuiTab-textColorPrimary MuiTab-labelIcon"
          id="menu-tab-0"
          role="tab"
          tabindex="-1"
          type="button"
        >
          <span
            class="MuiTab-wrapper"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 22c4.97 0 9-4.03 9-9-4.97 0-9 4.03-9 9zM5.6 10.25c0 1.38 1.12 2.5 2.5 2.5.53 0 1.01-.16 1.42-.44l-.02.19c0 1.38 1.12 2.5 2.5 2.5s2.5-1.12 2.5-2.5l-.02-.19c.4.28.89.44 1.42.44 1.38 0 2.5-1.12 2.5-2.5 0-1-.59-1.85-1.43-2.25.84-.4 1.43-1.25 1.43-2.25 0-1.38-1.12-2.5-2.5-2.5-.53 0-1.01.16-1.42.44l.02-.19C14.5 2.12 13.38 1 12 1S9.5 2.12 9.5 3.5l.02.19c-.4-.28-.89-.44-1.42-.44-1.38 0-2.5 1.12-2.5 2.5 0 1 .59 1.85 1.43 2.25-.84.4-1.43 1.25-1.43 2.25zM12 5.5c1.38 0 2.5 1.12 2.5 2.5s-1.12 2.5-2.5 2.5S9.5 9.38 9.5 8s1.12-2.5 2.5-2.5zM3 13c0 4.97 4.03 9 9 9 0-4.97-4.03-9-9-9z"
              />
            </svg>
            Support
          </span>
        </button>
      </div>
      <span
        class="PrivateTabIndicator-root-5 PrivateTabIndicator-colorSecondary-7 MuiTabs-indicator WithStyles(ForwardRef(Tabs))-indicator-2 PrivateTabIndicator-vertical-8"
        style="top: 0px; height: 0px;"
      />
    </div>
  </div>
</DocumentFragment>
`;
